devel/libtapi/Portfile

```diff
@@ -0,0 +1,109 @@
+                        sha256  50dafb7fa0c31e93b70e19a67e0f4a13f9dd9afc5dc507fd231a495edd7abdbd \
+                        size    37970663 \
+                        swift-4.2.4-RELEASE.tar.gz \
+                        rmd160  d83ffec2a76202da8e6ff0a994f222711bca49cc \
+                        sha256  74d27ed9f0371334de1ae9ad63d60869c60b6dca6ad6d7a39e6f2804e6007013 \
+                        size    17248048
+    post-extract {
+        file mkdir      ${workpath}/${worksrcdir}
+        file rename     ${workpath}/tapi-${version} ${worksrcpath}/tapi
+        file rename     ${workpath}/swift-llvm-swift-4.2.2-RELEASE ${worksrcpath}/llvm
+        file rename     ${workpath}/swift-clang-swift-4.2.4-RELEASE ${worksrcpath}/clang
+    }
+}
+
+variant tapillvm6 description "Build with llvm6 infrastructure" {
+    set llvm_version    6.0.1
+    master_sites-append https://github.com/llvm/llvm-project/archive/:llvm
+    distfiles-append    llvmorg-${llvm_version}.tar.gz:llvm
+    checksums-append    llvmorg-6.0.1.tar.gz \
+                        rmd160  3a934bace1f159adac75e5f5c0b01447561549bc \
+                        sha256  aefadceb231f4c195fe6d6cd3b1a010b269c8a22410f339b5a089c2e902aa177 \
+                        size    100123328
+    post-extract {
+        # we can't use ${llvm_version} here as error: "no such variable"
+        move ${workpath}/llvm-project-llvmorg-6.0.1  ${workpath}/${worksrcdir}
+        file rename ${workpath}/tapi-${version}      ${worksrcpath}/tapi
+        system -W ${worksrcpath} "ln -s ../../tapi   ${worksrcpath}/clang/tools/tapi"
+    }
+
+    patchfiles-append   patch-clang-tools-add-tapi.diff
+}
+
+variant tapillvm7 description "Build with llvm7 infrastructure, plus patches to add zippered TAPI and PPC support" {
+    set llvm_version    7.1.0
+    master_sites-append https://github.com/llvm/llvm-project/archive/:llvm
+    distfiles-append    llvmorg-${llvm_version}.tar.gz:llvm
+    checksums-append    llvmorg-7.1.0.tar.gz \
+                        rmd160  abdbeaa08cb1c5e9ce4008e47263eb10597f5aed \
+                        sha256  71c93979f20e01f1a1cc839a247945f556fa5e63abf2084e8468b238080fd839 \
+                        size    105306132
+    post-extract {
+        # we can't use ${llvm_version} here as error: "no such variable"
+        move ${workpath}/llvm-project-llvmorg-7.1.0 ${workpath}/${worksrcdir}
+        file rename ${workpath}/tapi-${version} ${worksrcpath}/tapi
+        system -W ${worksrcpath} "ln -s ../../tapi ${worksrcpath}/clang/tools/tapi"
+    }
+
+    # patches from https://github.com/iains/tapi to add zippered support, PPC support,
+    # build with llvm7, and adjust version numbering
+    patchfiles-append   patch-tapi-iains.diff
+    patchfiles-append   patch-clang-tools-add-tapi.diff
+}
+
+configure.args-append   -C ${worksrcpath}/tapi/cmake/caches/apple-tapi.cmake
+build.target            distribution
+destroot.target         install-distribution
+
+default_variants-append +tapillvm6
```

Comment:
It is possible that ```zippered``` support will be needed for 10.14+. If so, we'll have to make the default variant ```+tapillvm7``` for those systems.

We might also decide to just always go with @iains ' patches, and make the ```+tapillvm7``` variant the default for all systems. This has the advantage of some increased functionality, with the disadvantage of straying off the official beaten ```tapi + ld64``` path a little bit. But there is no word of a newer ```tapi``` drop coming any time soon, AFAIK.